src/librustc_back/svh.rs

```diff
@@ -51,19 +51,26 @@ use std::hash::{Hash, SipHasher, Hasher};
 use syntax::ast;
 use syntax::visit;
 
-#[derive(Clone, PartialEq, Debug)]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct Svh {
-    hash: String,
+    raw: u64,
 }
 
 impl Svh {
     pub fn new(hash: &str) -> Svh {
         assert!(hash.len() == 16);
-        Svh { hash: hash.to_string() }
-    }
+        // Ideally we'd just reverse the nibbles on LE machines during to_string, unfortunately
+        // this would break the abi so I guess we're just doing this now.
+
+        let s = if cfg!(target_endian = "big") {
+            hash.to_string()
+        } else {
+            hash.chars().rev().collect()
+        };
```

Comment:
Oh right. It needs to maintain compatibility with what's emitted today to avoid breaking the abi (without flipping it on LE arches it will refuse to build a stage1, because when it goes to consume the libc and core crates it'll believe they're out of date). I believe it's safe to nop out on BE architectures since they're already in the right order.